tests/Feature/SearchTest.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+/**
+ * We don't care for unhandled exceptions in tests.
+ * It is the nature of a test to throw an exception.
+ * Without this suppression we had 100+ Linter warning in this file which
+ * don't help anything.
+ *
+ * @noinspection PhpDocMissingThrowsInspection
+ * @noinspection PhpUnhandledExceptionInspection
+ */
+
+namespace Tests\Feature;
+
+use App\Facades\AccessControl;
+use Tests\Feature\Lib\AlbumsUnitTest;
+use Tests\Feature\Lib\PhotosUnitTest;
+use Tests\TestCase;
+
+class SearchTest extends TestCase
+{
+	/**
+	 * Test searching is working.
+	 *
+	 * @return void
+	 */
+	public function testSearch(): void
+	{
+		AccessControl::log_as_id(0);
+
+		$albums_test = new AlbumsUnitTest($this);
+		$photos_tests = new PhotosUnitTest($this);
+
+		/** @var string $id */
+		$id = $albums_test->add(null, 'search')->offsetGet('id');
+		/** @var string $tagId */
+		$tagId = $albums_test->addByTags('tag search', ['tag1', 'tag2'])->offsetGet('id');
+		$photoId = $photos_tests->upload(
+			TestCase::createUploadedFile(TestCase::SAMPLE_FILE_NIGHT_IMAGE)
+		);
+		$photos_tests->set_title($photoId, 'photo search');
+
+		$response = $this->postJson(
+			'/api/Search::run',
+			['term' => 'search']
+		);
+		$response->assertStatus(200);
+
+		$response->assertJson([
```

Comment:
This only works accidentally, because it implicitly assumes that there are no other photos and albums which match the search. This is indeed the case, because the other tests and test classes properly clean up after themselves.

In PR https://github.com/LycheeOrg/Lychee/pull/1351 I have added a special test class [`PhotoTestBase`](https://github.com/LycheeOrg/Lychee/blob/use_filestreams/tests/Feature/Base/PhotoTestBase.php) for that. I would like to wait with this PR until https://github.com/LycheeOrg/Lychee/pull/1351 has been merged and then this test class should extend `PhotoTestBase`.

Reply:
I fixed that in https://github.com/LycheeOrg/Lychee/pull/1395/commits/cde6d6326434225d5f8f1f091985d22a987cd5a2